text/FilipBartodziej/opis.tex

```diff
@@ -0,0 +1,188 @@
+\end{itemize}
+
+\end{remark}
+
+Przedstawmy również ważną własność kategoryzacji SL.
+
+\begin{lemma}\label{SL_order}
+Sufiks typu S jest leksykograficznie większy od każdego sufiksu typu L, który rozpoczyna się tym samym znakiem.
+\end{lemma}
+\begin{proof}
+Dowód nie wprost - załóżmy że istnieją sufiksy: $T_i = c \alpha c_1 \beta$ typ S i $T_j = c\alpha c_2 \gamma$ typ L, $c_1 \neq c_2$, takie że $T_i < T_j$.
+
+\item[Case 1:] $\alpha$ zawiera znak inny niż $c$\\
+Niech $c_3$ - znak $\neq c$ najbardziej na lewo w $\alpha$. $T_i$ jest typu S, więc $c_3>c$. Podobnie z $T_j$ typu L otrzymujemy $c_3<c$, sprzeczność.
+
+\item[Case 2:] $\alpha$ zawiera tylko $c$ lub jest puste\\
+Z typów $T_i$ i $T_j$ mamy $c_1 \geq c$ i $c \geq c_2$ (dowód podobnie jak case 1). Stąd $c_1 \geq c_2$, co daje sprzeczność z $T_i < T_j$.
+
+\end{proof}
+
+\section*{Algorytm small-large}
+
+Na algorytm można patrzeć w dwóch fazach - sortowanie sufiksów S/L, w zależności których z nich jest mniej. Następnie wykorzystujemy otrzymane informacje utworzenie porządku na wszystkich sufiksach. Wybieranie gałęzi z mniejszą ilością prefiksów danego typu jest konieczne aby zachować złożoność, ale nie wpływa na poprawność algorytmu.
+
+
+\begin{algorithmic}
+\Procedure{small-large}{$s$}\Comment{compute suffix array}
+\State $SL \gets SL\_categorization(s)$
+\If{$SL.count("S") < len(n)/2$}
+	\State $sortedS \gets sortS(s, SL)$
+	\State \Return $finishS(s, SL, sortedS)$
+\Else
+	\State $sortedL \gets sortL(s, SL)$
+	\State \Return $finishL(s, SL, sortedS)$
+\EndIf
+\EndProcedure
+\end{algorithmic}
+
+W kolejnych sekcjach opisane zostaną bardziej szczegółowo funkcje $sortS$ i $finishS$. Procedury dla typu L są bardzo podobne - należy przeglądać niektóre listy w odwrotnej kolejności, operacje przeniesienia na początek/koniec grupy zamienić na przeniesienie na koniec/początek i zastąpić odniesienia do typu S odniesieniami do typu L.
+
+\section*{Sortowanie S sufiksów - faza 1}
+
+\begin{definition}
+S odległość - dystans od i do najbliższego na lewo S sufiksu (nie wliczając i).
+\end{definition}
+
+\begin{definition}
+S podsłowo - podsłowo $t[i,i+1,...,j]$, gdzie i oraz j to S sufiksy, a pomiędzy są same L sufiksy.
+\end{definition}
+
+S podsłowa są ściśle powiązane z S sufiksami - gdy weźmiemy ciąg kolejnym S podsłów i skleimy je ze sobą, otrzymamy S sufiks modulo duplikaty znaków na zetknięciu podsłów.
+
+\begin{algorithmic}
+\Procedure{sortS}{s, SL}
+\State $SsubstrPrefixes \gets$ S substring prefixes, divided by Sdist, bucketed and sorted by first character
+\State $Ssubstr \gets$ S substrings, bucketed and sorted by first character
+\For{$d \gets 1,...,max\_Sdist$}
+\For{$j$ in $SsubstrPrefixes[d]$}
+\State move $j-d$ to the front of its group in $Ssubstr$
+\EndFor
+\State update groups in $Ssubstr$ using old $Ssubstr$ and $SsubstrPrefixes[d]$ 
+\EndFor
+\State $s' \gets$ map $Ssubstr$ to its buckets number, ordered as in $s$
+\State \Return $unmap(small-large(s'))$
```

Comment:
You have some overall font issues in pseudocodes e.g. here small-large should be small-caps to refer the reader to the previous pseudocode.

Reply:
I've switched to using "\Call" for function calls.